redis_json/src/commands.rs

```diff
@@ -540,21 +541,17 @@ pub fn prepare_paths_for_updating(paths: &mut Vec<Vec<String>>) {
     });
     // Remove paths which are nested by others (on each sub-tree only top most ancestor should be deleted)
     // (TODO: Add a mode in which the jsonpath selector will already skip nested paths)
-    let mut string_paths = Vec::new();
-    paths.iter().for_each(|v| {
-        string_paths.push(v.join(","));
-    });
+    let mut string_paths = paths.iter().map(|v| v.join(",")).collect::<Vec<_>>();
     string_paths.sort();
 
     paths.retain(|v| {
         let path = v.join(",");
-        let found = string_paths.binary_search(&path).unwrap();
-        for p in string_paths.iter().take(found) {
-            if path.starts_with(p.as_str()) {
-                return false;
-            }
-        }
-        true
+        let found = string_paths
```

Comment:
It was doing a linear scan up to that point regardless.

We always did the `path.starts_with(p)` on all the elements in `0..found`. But that must return true on `found` itself. We avoid binary searching for `found` and simply stop when we reach to any element, including the one at found, that matches the predicate.

Reply:
Then it sounds like we might mark the paths already seen so that we don't iterate over them and skip for further paths, as we surely will not need them.

Reply:
Maybe. It may be potentially worthwhile to remove the found path also from `string_paths` to avoid checking against it in the future. It is extra work either way.